### ISSUE TITLE
day archive fails due to DST transition 

### ISSUE THREAD

**Issue by SBFspot** [SBFspot/SBFspot#29]
The day archive fails after a DST switch.

The bug is found in 2.06 as well as in 2.2.11, the fix is for 2.2.11

the fix is as follows:
*** bug/SMAspot.cpp 2014-01-22 13:36:48.000000000 +0100
--- fix/SMAspot.cpp 2014-03-30 11:11:21.664330942 +0200

*** 576,581 ****
--- 576,583 ----
      puts("********************");
  }

startTime -= 86400;                   // to overcome problem with DST transition - RB@20140330

E_SMASPOT rc = E_OK;
  struct tm start_tm;
  memcpy(&start_tm, localtime(&startTime), sizeof(start_tm));

*** 583,588 ****
--- 585,591 ----
  start_tm.tm_hour = 0;
  start_tm.tm_min = 0;
  start_tm.tm_sec = 0;

start_tm.tm_mday += 1;                    // to overcome problem with DST transition - RB@20140330
  startTime = mktime(&start_tm);

  if (VERBOSE_NORMAL)

#### This work item was migrated from CodePlex
CodePlex work item ID: '23'
Assigned to: 'SBF'
Vote count: '2'

**Comment by SBFspot** [comment]
[SillieWimons@30/03/2014]
dear ruud,

the issue you reported is the same as issue 17 (by ron)

kr wim   (note i extend my information via issue17)

**Comment by SBFspot** [comment]
[SBF@31/03/2014]
@Ruud_B: thanks for the fix provided

**Comment by SBFspot** [comment]
[heino@1/04/2014]
This works with 2.2.10.

I think I understand what this fix does, but I don't have a clue why this works.

In the end I'm happy with it, thank you!

**Comment by SBFspot** [comment]
[jorispepijn@3/04/2014]
I have installed version 2.4.2 on my Raspberry Pi to get my 30/3 data out of my two converters but with no succes.  All data is correctly uploaded but 30/3 is scipped.

I noticed another problem on the Raspberry with version 2.4.2. Using putty to start SMA results in an error 

```
Commandline Args: -cfg/home/pvdiary2/bin/PVTimorstraat.cfg -ad8 -am0 -sp0 -finq
Unable to open 'date_time_zonespec.csv'
```

The 2.2.11 fix also didn't work in my case

**Comment by SBFspot** [comment]
[SBF@3/04/2014]
You can find 'date_time_zonespec.csv' in the source folder
Copy it to bin/Release (shoul be done by make)

**Comment by SBFspot** [comment]
[jorispepijn@3/04/2014]
dear SBF

Thanks for thinking with me

The file date_time_zonespec.csv is already in the bin/Release folder (and a copy in the source folder. The folder is only a problem when using Putty, which gives the error message. When using a Java script there is no problem concerning 'date_time_zonespec.csv' .  Only the 30/3 data isn't created.

**Comment by SBFspot** [comment]
[heino@3/04/2014]
I compiled 2.4.2 and it worked.
@jorisoeijn:
Are you sure you compiled succesfully? 
Please add puts("This is my compile"); , delete/rename the binary and try again.

**Comment by SBFspot** [comment]
[jorispepijn@4/04/2014]
Heino and SBF

I will do a new intalation on another raspberry and keep you informed

**Comment by SBFspot** [comment]
**Issue closed by _SBF_ with comment**
Fixed in V2.2.10

**Reason closed**
Fixed
